queries/zathurarc/highlights.scm

```diff
@@ -0,0 +1,30 @@
+; General
+(path) @module
+
+(option) @variable
+
+(command) @keyword
+
+(mode) @variable
+
+(mode_name) @type.builtin
+
+(key) @variable
+
+(key_name) @constant
+
+(function) @function.call
+
+(argument) @variable.parameter
+
+; Comments
+(comment) @comment.line @spell
+
+; Literals
+(int) @number
+
+(float) @number.float
```

Comment:
This doesn't work. `@number` will always be used. You probably need to change the parser so that `(float)` like `1.23` isn't parsed into `(float (int) "." (int))`

Reply:
So how to change it? I am not clear about your meaning. :cry:

Reply:
Because `1.23` is parsed into `(int) "." (int)` under the hood, `(int)`'s capture will be used, which is `@number`. Additionally, `(int)` is more specific than `(float)` in this case, so `(int)` will be prioritized

Reply:
You'll need to tweak the parser so that `(float)` don't get parsed to 2 ints with a dot.

Reply:
You means

```
set option 1.2
```

Should be parsed to

```
(set_directive) ; [4:1 - 14]
 (command) ; [4:1 - 3]
 (option) ; [4:5 - 10]
 (float) ; [4:12 - 14]
```

Not current

```
(set_directive) ; [4:1 - 14]
 (command) ; [4:1 - 3]
 (option) ; [4:5 - 10]
 (float) ; [4:12 - 14]
  (int) ; [4:12 - 12]
  (int) ; [4:14 - 14]
```

?

Reply:
Yes.

Reply:
OK, fixed.